multinet/api/tests/test_network.py

```diff
@@ -285,125 +269,106 @@ def test_network_rest_delete_unauthorized(workspace: Workspace, api_client: APIC
-        f'/api/workspaces/{workspace.name}/networks/{network.name}/nodes/',
-        nodes,
-    )
+    if success:
+        assert_limit_offset_results(
+            authenticated_api_client,
+            f'/api/workspaces/{workspace.name}/networks/{network.name}/nodes/',
+            nodes,
+        )
+    else:
+        r = r = authenticated_api_client.get(
+            f'/api/workspaces/{workspace.name}/networks/{network.name}/nodes/',
+            {'limit': 0, 'offset': 0},
+        )
+        assert r.status_code == status_code
 
 
 @pytest.mark.django_db
-def test_network_rest_retrieve_nodes_public(
-    public_workspace: Workspace, authenticated_api_client: APIClient
-):
+def test_network_rest_retrieve_nodes_public(public_workspace: Workspace, api_client: APIClient):
     network = populated_network(public_workspace)
     nodes = list(network.nodes())
 
     assert_limit_offset_results(
-        authenticated_api_client,
+        api_client,
         f'/api/workspaces/{public_workspace.name}/networks/{network.name}/nodes/',
         nodes,
     )
 
 
 @pytest.mark.django_db
-def test_network_rest_retrieve_nodes_no_access(
-    workspace: Workspace, authenticated_api_client: APIClient
-):
-    network = populated_network(workspace)
-    r = authenticated_api_client.get(
-        f'/api/workspaces/{workspace.name}/networks/{network.name}/nodes/',
-        {'limit': 0, 'offset': 0},
-    )
-    assert r.status_code == 404
-
-
-@pytest.mark.django_db
-@pytest.mark.parametrize('permission', workspace_role_range())
+@pytest.mark.parametrize(
+    'permission,is_owner,status_code,success',
+    [
+        (None, False, 404, False),
+        (WorkspaceRoleChoice.READER, False, 200, True),
+        (WorkspaceRoleChoice.WRITER, False, 200, True),
+        (WorkspaceRoleChoice.MAINTAINER, False, 200, True),
+        (None, True, 200, True),
+    ],
+)
 def test_network_rest_retrieve_edges(
     workspace: Workspace,
     user: User,
     authenticated_api_client: APIClient,
     permission: WorkspaceRoleChoice,
+    is_owner: bool,
+    status_code: int,
+    success: bool,
 ):
-    workspace.set_user_permission(user, permission)
+    if permission is not None:
+        workspace.set_user_permission(user, permission)
+    elif is_owner:
+        workspace.set_owner(user)
     network = populated_network(workspace)
     edges = list(network.edges())
 
-    assert_limit_offset_results(
-        authenticated_api_client,
-        f'/api/workspaces/{workspace.name}/networks/{network.name}/edges/',
-        edges,
-    )
+    if success:
+        assert_limit_offset_results(
+            authenticated_api_client,
+            f'/api/workspaces/{workspace.name}/networks/{network.name}/edges/',
+            edges,
+        )
+    else:
+        r = authenticated_api_client.get(
+            f'/api/workspaces/{workspace.name}/networks/{network.name}/edges/',
+            {'limit': 0, 'offset': 0},
+        )
+        assert r.status_code == 404
```

Comment:
If my above comment about omitting the `status_code` is applied, then this should stay. Otherwise, this should be changed to 
```suggestion
        assert r.status_code == status_code
```

Reply:
Keeping this as `404`, since I'm going with your suggestion.